### PR TITLE
Deanonymize fields

### PR DESCRIPTION
**Pull request by pepyakin**
r? @fitzgen

### CODE REVIEW

File: src/ir/context.rs [hunk 1]
```diff
@@ -799,6 +799,23 @@ impl BindgenContext {
         }
     }
 
+    /// Assign a new generated name for each anonymous field.
+    fn deanonymize_fields(&mut self) {
+        for item in self.items.values_mut() {
```
Comment (fitzgen) [inline src/ir/context.rs]: We should add `let _t = self.timer("deanonymize_fields");` at the top of this function.

File: src/ir/context.rs [hunk 2]
```diff
@@ -799,6 +799,23 @@ impl BindgenContext {
...
+            let ty = match item.kind_mut().as_type_mut() {
+                Some(ty) => ty,
+                None => continue,
+            };
+
+            let comp_info = match *ty.kind_mut() {
+                TypeKind::Comp(ref mut comp_info) => comp_info,
+                _ => continue,
```
Comment (fitzgen) [inline src/ir/context.rs]: This whole dance is a little more succinct as:

```rust
item.kind_mut()
    .as_type_mut()
    .and_then(|ty| ty.as_comp_mut())
    .map(|comp| comp.deanonymize_fields());
```

**Comment by pepyakin** [comment]
> Was also looking for the test expectation differences so that I could comment on them, see if they were serious or not, and maybe get an inkling for what might be triggering it.

So there [are none](https://travis-ci.org/rust-lang-nursery/rust-bindgen/builds/278735757?utm_source=github_status&utm_medium=notification)! 

If you would like to see what differences I'm encountered on my system LLVM [here it is](https://gist.github.com/pepyakin/9447c2a7f2ecb33979f8400ed7d2e79a).

**Comment by pepyakin** [comment]
> I'd expect that there would be some corresponding bits removed from codegen.

Sorry! Completely forgot about them.

File: src/ir/context.rs (hunk 2 above)
Comment (pepyakin) [reply src/ir/context.rs]: Is it really OK to have side effects inside a `map`?

File: src/ir/context.rs (hunk 1 above)
Comment (pepyakin) [reply src/ir/context.rs]: yeah sure!

File: src/ir/context.rs (hunk 2 above)
Comment (pepyakin) [reply src/ir/context.rs]: Anyway, iterators are lazy and should be consumed to provide side-effects. I see only two appropriate ways to consume Iterator here:

- `collect::<Vec<()>>`,
- `for_each`, which seems to be nightly

So I decided to do it a little bit differently and I hope you wouldn't mind!

Comment (fitzgen) [reply src/ir/context.rs]: Note that the `map` in my snippet was `Option::map`, not `Iterator::map`, and so it is not lazy. That approach works as well!

Comment (pepyakin) [reply src/ir/context.rs]: Ah, yeah! See it now, thanks.

**Comment by fitzgen** [comment]
I think that means rebase + squash :p

**Comment by pepyakin** [comment]
@fitzgen done!

**Pull request merged**